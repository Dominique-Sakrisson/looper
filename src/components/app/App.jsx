/* eslint-disable max-len */
import React from 'react';
import {Switch, Route, Link, BrowserRouter as Router} from 'react-router-dom';
import Header from './header/Header'
import TrackListPage  from '../../containers/TrackListPage/TrackListPage';
import RecordTrackPage from '../../containers/RecordTrackPage/RecordTrackPage';

export default function App() {
  return <>
    <Router>
      <Header />
      <Switch>
        <Route exact 
          path="/tracks"  

          component={TrackListPage} 
        />
        <Route exact 
          path="/record"  
          component={RecordTrackPage} 
        />
      </Switch>
    </Router>
  </>;
}
