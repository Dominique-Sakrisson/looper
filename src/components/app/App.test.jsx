/**
 * @jest-environment jsdom
 */
/* eslint-disable no-unused-vars */

import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { render, screen,  cleanup, waitFor } from '@testing-library/react';
import App from './App';
import * as Tone from 'tone';


jest.mock('./modules/AudioContext.js', () => ({
  polySynth: () => [{}]
}));

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<App />);
    console.log(asFragment());
    expect(asFragment()).toMatchSnapshot();
  });

  it('checks for keys', async () => {
    const { asFragment } = render(<App />);
    const keys = await screen.findAllByRole('button', {
      name: 'note-key'
    });
    expect(keys).toHaveLength(12);
    expect(asFragment()).toMatchSnapshot();
  });
});



