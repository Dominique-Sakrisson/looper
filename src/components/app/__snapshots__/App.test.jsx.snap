// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component checks for keys 1`] = `
<DocumentFragment>
  <button>
    Save Track
  </button>
  <ul
    class="trackNotes"
  >
    <li>
      Recordings
    </li>
  </ul>
  <button>
    Playback
  </button>
  <button>
    record
  </button>
  <div
    class="dark"
  />
  <img
    alt="need help?"
    src="/path/image"
  />
  <span>
    need help?
  </span>
  <br />
  <section
    class="settings"
  >
    <button
      aria-label="show-settings"
    >
       Show Settings 
    </button>
  </section>
  <div
    class="piano"
  >
    <div
      class="keyBoard"
    >
      <section
        class="keys"
      >
        <span>
          <button
            aria-label="note-key"
            class="keyButton c"
            role="button"
            value="c"
          >
            c
          </button>
        </span>
        <span>
          <button
            aria-label="note-key"
            class="keyButton d"
            role="button"
            value="d"
          >
            d
          </button>
        </span>
        <span>
          <button
            aria-label="note-key"
            class="keyButton e"
            role="button"
            value="e"
          >
            e
          </button>
        </span>
        <span>
          <button
            aria-label="note-key"
            class="keyButton f"
            role="button"
            value="f"
          >
            f
          </button>
        </span>
        <span>
          <button
            aria-label="note-key"
            class="keyButton g"
            role="button"
            value="g"
          >
            g
          </button>
        </span>
        <span>
          <button
            aria-label="note-key"
            class="keyButton a"
            role="button"
            value="a"
          >
            a
          </button>
        </span>
        <span>
          <button
            aria-label="note-key"
            class="keyButton b"
            role="button"
            value="b"
          >
            b
          </button>
        </span>
        <button
          aria-label="note-key"
          class="keyButtonSharp"
          role="button"
          value="c#"
        >
           c#
        </button>
        <button
          aria-label="note-key"
          class="keyButtonSharp"
          role="button"
          value="d#"
        >
           d#
        </button>
        <button
          aria-label="note-key"
          class="keyButtonSharp"
          role="button"
          value="f#"
        >
           f#
        </button>
        <button
          aria-label="note-key"
          class="keyButtonSharp"
          role="button"
          value="g#"
        >
           g#
        </button>
        <button
          aria-label="note-key"
          class="keyButtonSharp"
          role="button"
          value="a#"
        >
           a#
        </button>
      </section>
    </div>
  </div>
  <section
    class="chart"
  >
    <span
      style="z-index: 2; position: absolute; width: 5px; height: 200px; border: 1px solid black; left: 67px;"
    />
    &lt;div&gt;Loading Chart&lt;/div&gt;
  </section>
  <section
    class="pianoInstructions 
      // conditional to assign onScreen or OffScreen as well
      offScreen"
  >
    <img
      alt="need help?"
      src="/path/image"
    />
    <span>
      CLOSE
    </span>
    <div>
      <h3>
        How to record
      </h3>
      <p>
        (1)Hit record
      </p>
      <p>
        (2)Play notes
      </p>
      <p>
        (3)Hit record
      </p>
      <p>
        (4)Hit playback
      </p>
      <p>
        (4)Wait a few seconds
      </p>
      <h3>
        Has keyboard support!
      </h3>
      <p>
        Press keys c, d, e, f, g, a, b
      </p>
      <p>
        Press keys 1 - 5 to play sharp notes
      </p>
    </div>
  </section>
</DocumentFragment>
`;

exports[`App component renders App 1`] = `
<DocumentFragment>
  <ul
    class="header"
  >
    <a
      href="/tracks"
    >
      <li>
        Tracks
      </li>
    </a>
    <a
      href="/record"
    >
      <li>
        Record track 
      </li>
    </a>
    <a
      href="/challenge"
    >
      <li>
        Play Challenge
      </li>
    </a>
  </ul>
  <section
    class="sc-gtsrHT cdQWYq"
  >
    <h1>
      Welcome to the homepage
    </h1>
    <h3>
      (A work in progress)
    </h3>
    <img
      class="sc-dlnjwi jvXKti"
      src="/path/image"
    />
  </section>
</DocumentFragment>
`;
